src/t8_schemes/t8_scheme_mempool_cxx.hxx

```diff
@@ -0,0 +1,100 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_default_common_cxx.hxx
+ * We provide some functions that are useful across element classes.
+ */
+
+#ifndef T8_SCHEME_COMMON_CXX
+#define T8_SCHEME_COMMON_CXX
+
+#include <t8_element_cxx.hxx>
+#include <t8_element.h>
+#include <sc_functions.h>
+
+/* Macro to check whether a pointer (VAR) to a base class, comes from an
+ * implementation of a child class (TYPE). */
+#define T8_SCHEME_IS_TYPE(VAR, TYPE) ((dynamic_cast<TYPE> (VAR)) != NULL)
+
+class t8_scheme_mempool_c: public t8_eclass_scheme_c {
```

Comment:
```suggestion
class t8_eclass_scheme_with_mempool: public t8_eclass_scheme {
```
This is just a suggestion. Let me know if you have a different opinion on this.

Reply:
What is the meaning of the suffix `_c`? "common" or does it have something to do with the language "C"?